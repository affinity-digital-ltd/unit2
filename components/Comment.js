import { Component } from 'react'
import moment from 'moment'

class Comment extends Component {
  render () {
    const { comment } = this.props

    return <div className='c-article__comments'>
      <div className='card c-article__comment'>
        <div className='card-header'>
          <time dateTime={comment.created_at}>{moment(comment.created_at).format('LL')}</time>
        </div>
        <div className='card-body'>
          <h6 className='card-subtitle mb-2 text-muted'>
            {comment.name}
          </h6>
          {comment.body}
        </div>
      </div>
    </div>
  }
}

export default Comment
