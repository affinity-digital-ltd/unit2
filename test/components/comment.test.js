import Comment from '../../components/Comment'
jest.mock('../../lib/Auth')

describe('<Comment />', () => {
  it('it renders correctly', () => {
    const comment = {
      id: '1',
      name: 'Dave Smith',
      body: 'This is the **body** of my comment, which includes markdown',
      created_at: new Date(Date.UTC(97, 1, 2, 3, 4, 5))
    }

    const component = shallow(
      <Comment comment={comment} />
    )

    expect(component).toMatchSnapshot()
  })
})
