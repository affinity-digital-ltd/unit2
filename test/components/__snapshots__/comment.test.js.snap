// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> it renders correctly 1`] = `
<div
  className="c-article__comments"
>
  <div
    className="card c-article__comment"
  >
    <div
      className="card-header"
    >
      <time
        dateTime={1997-02-02T03:04:05.000Z}
      >
        February 2, 1997
      </time>
    </div>
    <div
      className="card-body"
    >
      <h6
        className="card-subtitle mb-2 text-muted"
      >
        Dave Smith
      </h6>
      This is the **body** of my comment, which includes markdown
    </div>
  </div>
</div>
`;
